src/ert/_c_wrappers/config/config_parser.py

```diff
@@ -9,13 +10,38 @@
 from ert._c_wrappers.config.unrecognized_enum import UnrecognizedEnum
 
 
+@dataclass()
+class Location:
+    filename: str
+    start_pos: Optional[int] = None
+    line: Optional[int] = None
+    column: Optional[int] = None
+    end_line: Optional[int] = None
+    end_column: Optional[int] = None
+    end_pos: Optional[int] = None
+
+
 class ConfigWarning(UserWarning):
     pass
 
 
 class ConfigValidationError(ValueError):
-    def __init__(self, errors: str, config_file: Optional[str] = None) -> None:
-        self.config_file = config_file
+    def __init__(
+        self,
+        errors: str,
+        config_file: Optional[str] = None,
+        location: Optional[Union[str, Location]] = None,
```

Comment:
I think with this design we can just use the type `Optional[Location]` instead of `Optional[Union[str, Location]]`